asciidtype/tests/test_asciidtype.py

```diff
@@ -10,11 +10,64 @@ def test_dtype_creation():
 
 def test_scalar_creation():
     dtype = ASCIIDType(7)
-    ASCIIScalar('string', dtype)
+    ASCIIScalar("string", dtype)
 
 
 def test_creation_with_explicit_dtype():
     dtype = ASCIIDType(7)
     arr = np.array(["hello", "this", "is", "an", "array"], dtype=dtype)
     assert repr(arr) == (
-        "array(['hello', 'this', 'is', 'an', 'array'], dtype=ASCIIDType(7))")
+        "array(['hello', 'this', 'is', 'an', 'array'], dtype=ASCIIDType(7))"
+    )
+
+
+def test_creation_truncation():
+    inp = ["hello", "this", "is", "an", "array"]
+
+    dtype = ASCIIDType(5)
+    arr = np.array(inp, dtype=dtype)
+    assert repr(arr) == (
+        "array(['hello', 'this', 'is', 'an', 'array'], dtype=ASCIIDType(5))"
+    )
+
+    dtype = ASCIIDType(4)
+    arr = np.array(inp, dtype=dtype)
+    assert repr(arr) == (
+        "array(['hell', 'this', 'is', 'an', 'arra'], dtype=ASCIIDType(4))"
+    )
+
+    dtype = ASCIIDType(1)
+    arr = np.array(inp, dtype=dtype)
+    assert repr(arr) == (
+        "array(['h', 't', 'i', 'a', 'a'], dtype=ASCIIDType(1))"
+    )
+    assert arr.tobytes() == b"htiaa"
+
+    # dtype = ASCIIDType()
+    # arr = np.array(["hello", "this", "is", "an", "array"], dtype=dtype)
+    # assert repr(arr) == ("array(['', '', '', '', ''], dtype=ASCIIDType(0))")
+    # assert arr.tobytes() == b""
```

Comment:
These tests are commented out because they depend on https://github.com/numpy/numpy/pull/22763 being merged in numpy.